testing/utils.go

```diff
@@ -23,3 +26,27 @@ func ApplyValSetChanges(tb testing.TB, valSet *tmtypes.ValidatorSet, valUpdates
 
 	return newVals
 }
+
+// VoteAndCheckProposalStatus votes on a gov proposal, checks if the proposal has passed, and returns an error if it has not with the failure reason.
+func VoteAndCheckProposalStatus(endpoint *Endpoint, proposalID uint64) error {
+	// vote on proposal
+	ctx := endpoint.Chain.GetContext()
+	require.NoError(endpoint.Chain.TB, endpoint.Chain.GetSimApp().GovKeeper.AddVote(ctx, proposalID, endpoint.Chain.SenderAccount.GetAddress(), govtypesv1.NewNonSplitVoteOption(govtypesv1.OptionYes), ""))
+
+	// fast forward the chain context to end the voting period
+	params, err := endpoint.Chain.GetSimApp().GovKeeper.Params.Get(ctx)
+	require.NoError(endpoint.Chain.TB, err)
+
+	endpoint.Chain.Coordinator.IncrementTimeBy(*params.VotingPeriod + *params.MaxDepositPeriod)
+	newHeader := endpoint.Chain.GetContext().BlockHeader()
+	ctx = ctx.WithBlockHeader(newHeader)
+	endpoint.Chain.NextBlock()
```

Comment:
won't `NextBlock()` update the context? I think the `newHeader` and `ctx.WithBlockHeader` is not needed here? 

You'd have to use `endpoint.Chain.GetContext()` again below, as the `ctx` var would be stale. 

Might be worth making this change

Reply:
yea I saw the same thing. The `NextBlock` is likely redundant, but I think this is an instance where it is nicer to just use `endpoint.Chain.GetContext` instead of assinging to a var